engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
```

Comment:
It looks strange to me haveing a method doing asynchronous calls to an actor inside of the DB layer to return an engine-level object type.
Is it not possible to change `getWorkflow` to `getExecutionAndAux` and have whoever is currently calling `getWorkflow` call `materializeWorkflowDescriptor` instead ?